parsers/NTESMO.py

```diff
@@ -0,0 +1,220 @@
+    "C4": Generator(power_plant="Channel Island Power Station", fuel_type="gas"),
+    "C5": Generator(power_plant="Channel Island Power Station", fuel_type="gas"),
+    "C6": Generator(power_plant="Channel Island Power Station", fuel_type="gas"),
+    "C7": Generator(power_plant="Channel Island Power Station", fuel_type="gas"),
+    "C8": Generator(power_plant="Channel Island Power Station", fuel_type="gas"),
+    "C9": Generator(power_plant="Channel Island Power Station", fuel_type="gas"),
+    "W1": Generator(power_plant="Weddell Power Station", fuel_type="gas"),
+    "W2": Generator(power_plant="Weddell Power Station", fuel_type="gas"),
+    "W3": Generator(power_plant="Weddell Power Station", fuel_type="gas"),
+    "LMS": Generator(
+        power_plant="LMS Energy from the Shoal Bay landfill", fuel_type="biomass"
+    ),
+    "K1": Generator(power_plant="Katherine Power Station", fuel_type="gas"),
+    "K2": Generator(power_plant="Katherine Power Station", fuel_type="gas"),
+    "K3": Generator(power_plant="Katherine Power Station", fuel_type="gas"),
+    "K4": Generator(power_plant="Katherine Power Station", fuel_type="gas"),
+    "P1": Generator(power_plant="Pine Creek Power Station", fuel_type="gas"),
+    "P2": Generator(power_plant="Pine Creek Power Station", fuel_type="gas"),
+    "P3": Generator(power_plant="Pine Creek Power Station", fuel_type="gas"),
+    "KS01": Generator(power_plant="", fuel_type="unknown"),
+    "MY01": Generator(power_plant="", fuel_type="unknown"),
+    "BJ01": Generator(power_plant="", fuel_type="unknown"),
+    "BP01": Generator(power_plant="", fuel_type="unknown"),
+    "HP01": Generator(power_plant="", fuel_type="unknown"),
+}
+
+
+def construct_year_index(year: int, session: Session) -> Dict[int, Dict[int, str]]:
+    """Browse all links on a yearly historical daily data and index them."""
+    index = {}
+    year_index_page = session.get(INDEX_URL.format(year))
+    soup = BeautifulSoup(year_index_page.text, "html.parser")
+    for month in range(1, 13):
+        index[month] = {}
+    for a in soup.find_all("a", href=True):
+        if a["href"].startswith("https://ntesmo.com.au/__data/assets/excel_doc/"):
+            date = pd.to_datetime(
+                a.find("div", {"class": "smp-tiles-article__title"}).text
+            )
+            index[date.month][date.day] = a["href"]
+    return index
+
+
+def get_historical_daily_data(link: str, session: Session) -> bytes:
+    result = session.get(link)
+    return result.content
+
+
+def extract_production_data(file: bytes) -> pd.DataFrame:
+    return pd.read_excel(
+        file, "Generating Unit Output", skiprows=4, header=0, usecols="A:AA"
+    )
+
+
+def extract_demand_price_data(file: bytes) -> pd.DataFrame:
+    return pd.read_excel(
+        file, "System Demand and Market Price", skiprows=4, header=0, usecols="A:C"
+    )
+
+
+def get_data(
+    session: Session,
+    target_datetime: Optional[datetime],
```

Comment:
```suggestion
    target_datetime: datetime = arrow.now().to("utc").shift(hours=-DELAY),
```

OR something like that

Reply:
actually it should probably be done at the `fetch_*` level

Reply:
Agree, actually it would be better at fetch_ level.

Reply:
I would suggest using `datetime.now() - timedelta(hours=30)` but otherwise it makes sense.